### ISSUE TITLE
Natural Weapon Stacking does not appear to work properly with Dragon Bloodline claws

### ISSUE THREAD

**Issue by pheonix99** [microsoftenator2022/MiscTweaksAndFixes#1]
Gave this a quick test (TTT-Base disabled because of ongoing bugs) and there's no stacking - elemental riders or size up.

Getting it to work with claws from the same element bloodline or the ones before the element shows up could be finicky and required dedicated logic because those are the same weapon "item" and enchantment but it isn't working cross-element with the full power ones either which shouldn't have any special behavior?

**Comment by microsoftenator2022** [comment]
Interesting. I think this may be a vanilla bug with second bloodline. There only seems to be one set of (dragon) claws present despite having two dragon bloodlines. It might also be affected by my primalist bloodline changes (although claws shouldn't be touched by that).

Some testing results for reference

![image](https://user-images.githubusercontent.com/105488202/214822442-23de0283-cc6c-4b63-afd3-35d059979b3f.png)

```
[MiscTweaksAndFixes] [DEBUG] ItemEntityWeapon_get_Size_Patch.Postfix
[MiscTweaksAndFixes] [DEBUG] Blueprint: Claw - 484ea613eeaf2624fbb1a456302730a4
[MiscTweaksAndFixes] [DEBUG] Type: Claw
[MiscTweaksAndFixes] [DEBUG] Wielder: asdf
[MiscTweaksAndFixes] [DEBUG] EmptyHandOverride: 2, AddSecondaryAttacks: 1, AddAdditionalLimb: 1
[MiscTweaksAndFixes] [DEBUG] Empty hand overrides:
[MiscTweaksAndFixes] [DEBUG]      - 282c65d891dbcfa4dbcc779207d4db5c
[MiscTweaksAndFixes] [DEBUG]       * Magic Corrosive Claw (Claw) - 2e4ea327ad2330d40807b0d9c3492896
[MiscTweaksAndFixes] [DEBUG]      - 80ab616e15509f341bc4b42c2c480499
[MiscTweaksAndFixes] [DEBUG]       * Claw (Claw) - 484ea613eeaf2624fbb1a456302730a4
[MiscTweaksAndFixes] [DEBUG] Secondary attacks:
[MiscTweaksAndFixes] [DEBUG]      - a67b51a8074ae47438280be0a87b01b6
[MiscTweaksAndFixes] [DEBUG]         Bite (Bite) - 8c01e7fccbb829947bc5894f63fb389a
[MiscTweaksAndFixes] [DEBUG] Additional Limbs:
[MiscTweaksAndFixes] [DEBUG]     Dragon Bite - 572329fc0af467a43847d39f8c6162ea
[MiscTweaksAndFixes] [DEBUG]         Dragon Bite (Bite) - c66afbc07845e4245bf62021b7278a43
[MiscTweaksAndFixes] [DEBUG] Size change = 1: 1d8 (Medium) -> 2d6 (Large)
[MiscTweaksAndFixes] [DEBUG] RuleAttackWithWeapon_OnTrigger_Patch.Postfix
[MiscTweaksAndFixes] [DEBUG] Removing enchant  '631cb72d11015374987c161a2451a1cf'
[MiscTweaksAndFixes] [DEBUG] Removing enchant  '633b38ff1d11de64a91d490c683ab1c8'
[MiscTweaksAndFixes] [DEBUG] RuleAttackWithWeapon_OnTrigger_Patch.Prefix
[MiscTweaksAndFixes] [DEBUG] Adding enchant Magic Weapon '631cb72d11015374987c161a2451a1cf' from weapon Magic Corrosive Claw'2e4ea327ad2330d40807b0d9c3492896'
[MiscTweaksAndFixes] [DEBUG] Adding enchant Corrosive '633b38ff1d11de64a91d490c683ab1c8' from weapon Magic Corrosive Claw'2e4ea327ad2330d40807b0d9c3492896'
```
I don't actually know what the specific cause is but it looks like just checking the list of "equipped" natural weapons and buffs isn't enough

**Comment by microsoftenator2022** [comment]
Multiple instances of the same weapon blueprint across multiple bloodlines should be fixable, but it will probably be messy.

**Comment by microsoftenator2022** [comment]
For bloodrager, at least part of the problem is that claws are added by a big `if... else if ... else if ...` chain so it only adds the *first* matching set of claws.

**Comment by pheonix99** [comment]
Oh yeah, Bloodrager claws just plain old needs a refactor to work with this setup.

Also, natural weapon blueprints gets reused so often by the game and modders that I don't think you can have multiple instances of the same weapon be ignored safely? Preventing the same buff from being stacked is alreeady handled at the buff layer.

**Comment by microsoftenator2022** [comment]
I already check for different buffs, so the issue is mostly just the way those buffs work, which *is* solvable, just with slightly more sophisticated logic

**Comment by microsoftenator2022** [comment]
Should be fixed in 0.9.6

I had to make a special case for bloodrager so hopefully there's no other cases like it.
